ParadiseSS13/Paradise
# Adds code enforcement for rules regarding disabilities and jobs

## What Does This PR Do
Revives PR #20338 . Almost all of the description will be an exact copy paste of the original PR. This is the same code, as nothing of the original PR seems to have gone out of date, so no changes in code have been made. Let's pray I don't get stuck in squash hell. A lot of thanks go out to Miraviel for both making the code, and allowing me to revive it!
Original post, go:
We have server rules (and head of staff rulings) on whether certain jobs, such as command, VIP, or security, can have certain disabilities and missing limbs or not. So far it was not enforced by code, now it is.

TLDR You cannot be command, VIP, or security if you possess certain disabilities rendering you unable to do your job properly, and you cannot be combat-oriented roles (security, blueshield) or the CE if you lack limbs (prosthetics are ok). Both cases are covered by server rule 5. See the chart below for the exact restrictions.

**If you do not agree with this ruling, bring it to the forums for discussion, this PR is not changing any server rules, only enforces them.**
![image](https://github.com/ParadiseSS13/Paradise/assets/108773801/bd2f8975-81a1-4504-894c-d97dbcc5db82)





## Why It's Good For The Game
Less work for us to enforce this, less confusion and bad feelings on the players' side when they come up with a character concept, the game lets them create said character, then we bwoink them.


## Images of changes
Setting "Blind":
![image](https://github.com/ParadiseSS13/Paradise/assets/108773801/34552691-0a40-4911-a911-6d4b1139a9c4)

Setting "Mute":
![image](https://github.com/ParadiseSS13/Paradise/assets/108773801/0837eed1-15c8-4931-bc35-423b56c1d41e)

Setting left foot to "Amputated":
![image](https://github.com/ParadiseSS13/Paradise/assets/108773801/8c30dd2e-d19c-4cbf-93ff-920a0cfdce1b)



## Testing
(Okay, act like these lists are numbered correctly)

1. Open character setup
2. Set a disability according to the table
3. Look at job selection, it is denied
--

1. Open character setup
2. Select a job that has these restrictions
3. Set a blacklisted disability
4. Try to spawn in
5. (You won't)
--

1. Open character setup
2. Set a disability
3. Save
4. Try to latejoin in with a job that has it blacklisted

Sidenote: yes, I checked this all again to see if it still worked correctly.


## Changelog
:cl: Miraviel, DGamerL 
add: Code support was added (read: you can no longer save or spawn in with such a character) to something we have been enforcing via server rules for years now: disabilities and command/security roles. Security, Blueshield, and CE cannot be genetically blind, mute, deaf, dizzy, or have any amputated limbs (prosthetics are ok). Department heads cannot be blind, mute, deaf, or dizzy. VIPs (magistrate, IAA, NTrep) cannot be mute or deaf.
/:cl:
